Create the Boa v0.17 release post (#94)
boa-dev/boa-dev.github.io: posts/2023-07-05-boa-release-17.md

are two provided macros: `js_string!` for creating a new `JsString` from a `&str`, and `utf16!` for creating a utf-16
array literal from a `&str`.

You can create a utf-16 array literal from any utf-8 `str`.

```rust
const HELLO: &[u16] = utf16!("Hi! :)");
```

You can create a `JsString` from a string literal with the `js_string` macro.

```rust
let hw = js_string!("Hello, world!");
assert_eq!(&hw, utf16!("Hello, world!"));
```

You can also pass any number of `&[u16]` string values as arguments to create a new `JsString`.

```rust
const NAME: &[u16]  = utf16!("human! ");
let greeting = js_string!("Hello, ");
let msg = js_string!(&greeting, &NAME, utf16!("Nice to meet you!"));

assert_eq!(&msg, utf16!("Hello, human! Nice to meet you!"));
```

## Conclusions

If you reached so far, you probably understand how big this release was, and you can find even more changes in the
[full changelog][changelog]. Boa is now becoming a real option for many projects, which shows with the amount of
financial support we have received these last months. Nevertheless, going forward, we need your help to get to a 1.0

> version. Wheter you are good with Rust, JavaScript, documentation or development, we have multiple

Reviewer: ```suggestion
version. Whether you are good with Rust, JavaScript, documentation or development, we have multiple
```

[good first issues][first_issues], and places where we [need help][help_wanted], both in Boa's
[main repository][boa_engine] and [others][boa_org] around it.

> Once again, we have to thank to [all the contributors][contributors] of this release!!

Reviewer: ```suggestion
Once again, big thanks to [all the contributors][contributors] of this release!!
```
